cmd/manager/main.go

```diff
@@ -217,88 +219,120 @@ func (s *server) SetEarlyStoppingParameters(ctx context.Context, in *pb.SetEarly
+func (s *server) SaveStudy(ctx context.Context, in *api_pb.SaveStudyRequest) (*api_pb.SaveStudyReply, error) {
 	var err error
 	if s.msIf != nil {
 		err = s.msIf.SaveStudy(in)
 	}
-	return &pb.SaveStudyReply{}, err
+	return &api_pb.SaveStudyReply{}, err
 }
 
-func (s *server) SaveModel(ctx context.Context, in *pb.SaveModelRequest) (*pb.SaveModelReply, error) {
+func (s *server) SaveModel(ctx context.Context, in *api_pb.SaveModelRequest) (*api_pb.SaveModelReply, error) {
 	if s.msIf != nil {
 		err := s.msIf.SaveModel(in)
 		if err != nil {
 			log.Printf("Save Model failed %v", err)
-			return &pb.SaveModelReply{}, err
+			return &api_pb.SaveModelReply{}, err
 		}
 	}
-	return &pb.SaveModelReply{}, nil
+	return &api_pb.SaveModelReply{}, nil
 }
 
-func (s *server) GetSavedStudies(ctx context.Context, in *pb.GetSavedStudiesRequest) (*pb.GetSavedStudiesReply, error) {
-	ret := []*pb.StudyOverview{}
+func (s *server) GetSavedStudies(ctx context.Context, in *api_pb.GetSavedStudiesRequest) (*api_pb.GetSavedStudiesReply, error) {
+	ret := []*api_pb.StudyOverview{}
 	var err error
 	if s.msIf != nil {
 		ret, err = s.msIf.GetSavedStudies()
 	}
-	return &pb.GetSavedStudiesReply{Studies: ret}, err
+	return &api_pb.GetSavedStudiesReply{Studies: ret}, err
 }
 
-func (s *server) GetSavedModels(ctx context.Context, in *pb.GetSavedModelsRequest) (*pb.GetSavedModelsReply, error) {
-	ret := []*pb.ModelInfo{}
+func (s *server) GetSavedModels(ctx context.Context, in *api_pb.GetSavedModelsRequest) (*api_pb.GetSavedModelsReply, error) {
+	ret := []*api_pb.ModelInfo{}
 	var err error
 	if s.msIf != nil {
 		ret, err = s.msIf.GetSavedModels(in)
 	}
-	return &pb.GetSavedModelsReply{Models: ret}, err
+	return &api_pb.GetSavedModelsReply{Models: ret}, err
 }
 
-func (s *server) GetSavedModel(ctx context.Context, in *pb.GetSavedModelRequest) (*pb.GetSavedModelReply, error) {
-	var ret *pb.ModelInfo = nil
+func (s *server) GetSavedModel(ctx context.Context, in *api_pb.GetSavedModelRequest) (*api_pb.GetSavedModelReply, error) {
+	var ret *api_pb.ModelInfo = nil
 	var err error
 	if s.msIf != nil {
 		ret, err = s.msIf.GetSavedModel(in)
 	}
-	return &pb.GetSavedModelReply{Model: ret}, err
+	return &api_pb.GetSavedModelReply{Model: ret}, err
+}
+
+func (s *server) Check(ctx context.Context, in *health_pb.HealthCheckRequest) (*health_pb.HealthCheckResponse, error) {
+	resp := health_pb.HealthCheckResponse{
+		Status: health_pb.HealthCheckResponse_SERVING,
+	}
+
+	// We only accept optional service name only if it's set to suggested format.
+	if in != nil && in.Service != "grpc.health.v1.Health" {
+		resp.Status = health_pb.HealthCheckResponse_UNKNOWN
+		return &resp, fmt.Errorf("grpc.health.v1.Health can only be accepted if you specify service name.")
+	}
+
+	// Check if we can connect to vizier-db since otherwise manager could not serve most of its methods.
+	dbIf, err := kdb.New()
```

Comment:
You should check dbIf is already initialized or not since another APIs will be called asynchronously.
You should not rewrite dbIf when it has been used in another function.

Reply:
Indeed, thanks! Will push a fix shortly.